ramses_rf/protocol/parsers.py

```diff
@@ -2629,18 +2643,70 @@ def parser_4401(payload, msg) -> dict:
 
 @parser_decorator  # hvac_4e01
 def parser_4e01(payload, msg) -> dict:
-    return {f"val_{x}": temp_from_hex(payload[x : x + 4]) for x in range(2, 34, 4)}
+    return {
+        f"temperature_{idx}": temp_from_hex(payload[x : x + 4])
+        for idx, x in enumerate(range(2, 34, 4), 1)
+        if payload[x : x + 4] != "7FFF"
+    }
 
 
 @parser_decorator  # hvac_4e02
 def parser_4e02(payload, msg) -> dict:
+    # UFC (itho autotemp slave to master, reporting thermostat setpoints and mode)
+    # separate message per thermostat (autotemp channel)
+    # heating:
+    #  I --- 02:248945 02:250708 --:------ 4E02 034 00-7FFF-07D0-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-04-7FFF-0834-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF
+    #  I --- 02:248945 02:250708 --:------ 4E02 034 00-07D0-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-04-0834-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF
+    # cooling:
+    #  I --- 02:248945 02:250708 --:------ 4E02 034 00-07D0-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-02-0834-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF-7FFF
+
+    if payload[34:36] == "02":
+        mode = "cooling"
+    elif payload[34:36] == "04":
+        mode = "heating"
+    else:
+        mode = "unknown"
+
     return (
-        {f"val_{x}": temp_from_hex(payload[x : x + 4]) for x in range(2, 34, 4)}
-        | {"val_34": payload[34:36]}
-        | {f"val_{x}": temp_from_hex(payload[x : x + 4]) for x in range(36, 68, 4)}
+        {
+            f"heating_setpoint_{idx}": temp_from_hex(payload[x : x + 4])
+            for idx, x in enumerate(range(2, 34, 4), 1)
+            if payload[x : x + 4] != "7FFF"
+        }
+        | {"mode": mode}
+        | {
+            f"cooling_setpoint_{idx}": temp_from_hex(payload[x : x + 4])
+            for idx, x in enumerate(range(36, 68, 4), 1)
+            if payload[x : x + 4] != "7FFF"
+        }
     )
 
 
+@parser_decorator  # hvac_4e0d
+def parser_4e0d(payload, msg) -> dict:
+    #  I --- 02:250704 02:250984 --:------ 4E0D 002 0100
+    #  I --- 02:250704 02:250984 --:------ 4E0D 002 0101
```

Comment:
No... I only see `4E0D` with 01 (`0100`) or (`0101`) are the only two flavours I have seen.

The only pattern I see, is that they are ONLY sent from autotemp master to slave. Never from slave to master.

This is really all "unknown"

Reply:
OK, ignoring the context for now, it seems to be a mode? I have only this transition:
```text
2022-08-21T23:28:47.127763 045  I --- 02:250704 02:250984 --:------ 4E0D 002 0101
2023-03-19T00:24:29.588078 053  I --- 02:250704 02:250984 --:------ 4E0D 002 0100
```

That is, it was `01` in Summer, and `00` in Winter (I have no packets in between).

Do you have more information than that?

Reply:
@tomkooij It would help me if you could confirm your `known_list`.

I am particularly interested in confirming which is spider autotemp/master/slave/thermostat, and which are fans